webapp/src/types/mattermost-webapp/index.d.ts

```diff
@@ -1,7 +1,42 @@
+export type BasePluginConfigurationSetting = {
+    name: string;
+    title?: string;
+    helpText?: string;
+    default?: string;
+}
+
+export type PluginConfigurationRadioSettingOption = {
+    value: string;
+    text: string;
+    helpText?: string;
+}
+
+export type PluginConfigurationRadioSetting = BasePluginConfigurationSetting & {
+    type: 'radio';
+    default: string;
+    options: PluginConfigurationRadioSettingOption[];
+}
+
+export type PluginConfigurationSetting = PluginConfigurationRadioSetting
+
+export type PluginConfigurationSection = {
+    settings: PluginConfigurationSetting[];
+    title: string;
+    onSubmit?: (changes: {[name: string]: string}) => void;
```

Comment:
```suggestion
    onSubmit?: (changes: Record<string, string>) => void;
```

Reply:
I feel this is more intention revealing, since you can name the key. But 0/5.